backpack/core/derivatives/crossentropyloss.py

```diff
@@ -26,8 +26,14 @@ def sqrt_hessian(self, module, g_inp, g_out):
 
         return sqrt_H
 
-    def sqrt_hessian_sampled(self, module, g_inp, g_out):
-        M = self.MC_SAMPLES
+    def make_sqrt_hessian_sampled_fn(self, mc_samples=1):
+        def _sqrt_hessian_sampled_fn(module, g_inp, g_out):
+            return self.sqrt_hessian_sampled(module, g_inp, g_out, mc_samples)
+
+        return _sqrt_hessian_sampled_fn
```

Comment:
This function seems overly complex. A [`partial evaluation`](https://docs.python.org/3/library/functools.html#functools.partial) for the specialization of the `mc_sample` parameter seems simpler.
```
def make_sqrt_hessian_sampled_fn(self, mc_samples=1):
    return partial(self.sqrt_hessian_sampled, mc_samples=mc_samples)
````